send-report.sh

```diff
@@ -0,0 +1,250 @@
+                    user="$OPTARG"
+                    ;;
+                p)
+                    password="$OPTARG"
+                    ;;
+                i)
+                    issuer_url="$OPTARG"
+                    ;;
+                c)
+                    client_id="$OPTARG"
+                    ;;
+                s)
+                    client_secret="$OPTARG"
+                    ;;
+                *)
+                    echo "Unknown option: -$OPTARG"
+                    usage
+                    ;;
+            esac
+        done
+}
+
+get_matching_identifier_refs() {
+    auth="$1"
+    if [ "$auth" == "no-auth" ]; then
+        response=$(blazectl download --server "$report_server" DocumentReference \
+                                              --query "identifier=$proj_ident_sys|$proj_ident_val")
+        echo "$response"
+    elif [ "$auth" == "basic-auth" ]; then
+        response=$(blazectl download --server "$report_server" DocumentReference \
+                                     --query "identifier=$proj_ident_sys|$proj_ident_val"\
+                                     --user "$user" --password "$password")
+        echo "$response"
+    else
+        oauth_response=$(curl -s -X POST "$issuer_url" \
+                            -H "Content-Type: application/x-www-form-urlencoded" \
+                            -d "grant_type=client_credentials" \
+                            -d "client_id=$client_id" \
+                            -d "client_secret=$client_secret")
+        fhir_report_bearer_token=$(echo "$oauth_response" | jq -r '.access_token')
+
+        response=$(blazectl download --server "$report_server" DocumentReference \
+                                             --query "identifier=$proj_ident_sys|$proj_ident_val"\
+                                             --token "$fhir_report_bearer_token")
+        echo "$response"
+    fi
+}
+
+get_doc_ref_id() {
+    auth="$1"
+    matching_refs=$(get_matching_identifier_refs "$auth")
+
+    count=$(echo "$matching_refs" | jq -s 'length')
+
+    if [[ "$count" -gt 1 ]]; then
+        echo "Error: Multiple DocumentReferences exist for masterIdentifier {system: '$proj_ident_sys', value: '$proj_ident_val'}"
+        exit 1
+    elif [[ "$count" -eq 1 ]]; then
+        doc_id=$(echo "$matching_refs" | jq -r '.id')
+        echo "$doc_id"
+    else
+        echo ""
+    fi
+}
+
+create_bundle() {
+    auth="$1"
+    if ! doc_ref_id=$(get_doc_ref_id "$auth")
+    then
+      echo "$doc_ref_id"
+      exit 1
+    fi
+
+    doc_ref_url="01"
+    report_url="02"
+
+    tz=$(date +"%z")  # e.g. "+0100"
+    tz=$(echo "$tz" | sed -E 's/^([+-][0-9]{2})([0-9]{2})$/\1:\2/') # +0100 -> +01:00
+    # -> this can be done with %:z, but '%:z' is not supported at all versions of the date command
+
+    date="$(date +"%Y-%m-%dT%H:%M:%S")${tz}"
+
+    bundle=$(bundle_template | jq -c --slurpfile report "$report_file" '
+      .entry[0].resource.date = $date |
```

Comment:
I ment, you can just use the bundle function to build the whole bundle like here: https://github.com/samply/blaze/blob/d7db28cc0602f590b0963c2087398544d9d6d8a6/docs/performance/synthea/create-medication-references.sh#L5

I really like jq, but that looks bloated.